backend/src/controllers/v2/secretsController.ts

```diff
@@ -621,172 +642,180 @@ export const getSecrets = async (req: Request, res: Response) => {
-                { user: { $exists: false } } // shared secrets from workspace
-            ]
-        }
-
-        if (normalizedPath == ROOT_FOLDER_PATH) {
-            secretQuery.path = { $in: [ROOT_FOLDER_PATH, null, undefined] }
-        } else if (normalizedPath) {
-            secretQuery.path = normalizedPath
-        }
-
-        if (tagIds.length > 0) {
-            secretQuery.tags = { $in: tagIds };
-        }
-
-        if (hasWriteOnlyAccess) {
-            // only return the secret keys and not the values since user does not have right to see values
-            secrets = await Secret.find(secretQuery).select("secretKeyCiphertext secretKeyIV secretKeyTag").populate("tags")
-        } else {
-            secrets = await Secret.find(secretQuery).populate("tags")
-        }
+    folderId = folder.id;
+  }
+
+  // secrets to return
+  let secrets: ISecret[] = [];
+
+  // query tags table to get all tags ids for the tag names for the given workspace
+  let tagIds = [];
+  const tagNamesList =
+    typeof tagSlugs === "string" && tagSlugs !== "" ? tagSlugs.split(",") : [];
+  if (tagNamesList != undefined && tagNamesList.length != 0) {
+    const workspaceFromDB = await Tag.find({ workspace: workspaceId });
+    tagIds = _.map(tagNamesList, (tagName: string) => {
+      const tag = _.find(workspaceFromDB, { slug: tagName });
+      return tag ? tag.id : null;
+    });
+  }
+
+  if (req.user) {
+    // case: client authorization is via JWT
+    const hasWriteOnlyAccess = await userHasWriteOnlyAbility(
+      req.user._id,
+      new Types.ObjectId(workspaceId),
+      environment
+    );
+    const hasNoAccess = await userHasNoAbility(
+      req.user._id,
+      new Types.ObjectId(workspaceId),
+      environment
+    );
+    if (hasNoAccess) {
+      throw UnauthorizedRequestError({
+        message:
+          "You do not have the necessary permission(s) perform this action",
+      });
     }
 
-    // case: client authorization is via service token
-    if (req.serviceTokenData) {
-        const userId = req.serviceTokenData.user;
-
-        const secretQuery: any = {
-            workspace: workspaceId,
-            environment,
-            $or: [
-                { user: userId }, // personal secrets for this user
-                { user: { $exists: false } } // shared secrets from workspace
-            ]
-        }
-
-        // TODO: check if user can query for given path
-        if (normalizedPath == ROOT_FOLDER_PATH) {
-            secretQuery.path = { $in: [ROOT_FOLDER_PATH, null, undefined] }
-        } else if (normalizedPath) {
-            secretQuery.path = normalizedPath
-        }
-
-        if (tagIds.length > 0) {
-            secretQuery.tags = { $in: tagIds };
-        }
-
-        // TODO check if service token has write only permission 
+    const secretQuery: any = {
+      workspace: workspaceId,
+      environment,
+      folder: folderId,
```

Comment:
Double checking, are we sticking to "root" as folder id or undefined for root? It looks like we have a mix of both in the code base

Reply:
Just saw the `backfillSecretFolders` and looks like you are back filling eveything to root if they don't already have a folder